src/ripple/shamap/SHAMap.h

```diff
@@ -82,7 +82,7 @@ class SHAMap
     Family&                         f_;
     beast::Journal                  journal_;
     std::uint32_t                   seq_;
-    std::uint32_t                   ledgerSeq_; // sequence number of ledger this is part of
+    std::uint32_t mutable           ledgerSeq_; // sequence number of ledger this is part of
```

Comment:
I don't get how it can work without `mutable` on `ledgerSeq_`: the value is modified from `fetchNodeFromDB` as you say. It'd be great if we could just abstract this functionality out of `SHAMap` but that's a much bigger change that I don't want to incorporate into this function.